mdoc/Mono.Documentation/Updater/DocumentationEnumerator.cs

```diff
@@ -27,8 +27,9 @@ protected IEnumerable<TypeDefinition> GetDocumentationTypes (AssemblyDefinition
                 if (seen != null && seen.Contains (type.FullName))
                     continue;
                 yield return type;
-                foreach (TypeDefinition nested in type.NestedTypes)
-                    yield return nested;
+                // because of this code we get nested classes twice
+//                foreach (TypeDefinition nested in type.NestedTypes)
```

Comment:
Commenting out code like this makes me a bit worried because it _seems_ unrelated to the other changes.

- Was there another change added that compensates for this code being commented out?
- Why weren't we seeing double nested classes before if this code was present in the codebase?
- Do we not have any nested types in our integration tests?
    - just checked, yeah, we do in `DocTest.cs` ... for example, `Widget.NestedClass`.

I'd like a bit more investigation around this code before removing it ... does it only affect the F# nested types? What happens if you add it back ... in particular, which integration tests fail. What happens if you comment out this code in the `master` branch and then run the integration tests?

Reply:
Before answering the questions, I'd like to describe this issue in general.
When I mention double nested classes I mean this method returns nested classes twice, they don't appear twice in the result XML file. The reason why this behavior is hidden is MDocUpdater class. Usually, it processes double values correctly, first occurrence is processed as a new value, and the second occurrence refreshes already generated nodes with the same value. So, it doesn't affect the result XML, but there are excess operations. 
In F# I encountered this issue because there are extension methods in a nested class in the integration test. In this case, MDocUpdater doesn't process twice iterated members correctly.

> Was there another change added that compensates for this code being commented out?
> Why weren't we seeing double nested classes before if this code was present in the codebase?

There were no other changes to compensate this one. We weren't seeing double nested classes because this case is handled by MDocUpdater class as I described above.

> Do we not have any nested types in our integration tests? just checked, yeah, we do in DocTest.cs ... for example, Widget.NestedClass.

> does it only affect the F# nested types? 

It affects all input assemblies and formatters.

> What happens if you add it back ... in particular, which integration tests fail. 
 
check-monodocer-fsharp fails with "Failed to compare two elements in the array." The issue appears because there are extension methods in a nested class.

> What happens if you comment out this code in the master branch and then run the integration tests?

Integration tests work as expected.

Reply:
Ahh I see ... excellent work rooting out the erroneous code :) Please remove it from the code entirely, as it doesn't need to stay commented out.